### DISCUSSION TITLE
Installation of MetPlus on Windows 10 WSL2 & AWS Cloud

### DISCUSSION THREAD

**Discussion by marstonsward** [dtcenter/METplus#1168] [Installation] [answered]
Hi,

# Use Case 
My company would like to begin using MetPlus. I would like to first test it on one of our windows PC using WSL2. Later on our cloud team would like to move MetPlus to the cloud. I see that there is a docker image and I wondering if it can be used as a devcontainer in VS Code - if anyone has experience with that. 

# Use Case Name and Category 
Installation in new environments

# Input Data ###
Synthetic vertically integrated liquid and dBZ

**Comment by JohnHalleyGotway** [comment]
Converting this support question from an issue to a discussion. Please post questions as [new discussions](https://github.com/dtcenter/METplus/discussions/new).

**Comment by georgemccabe** [comment]
Hi @Marston,

I have installed METplus on a Windows PC in WSL2. To do this I first installed the MET executables following the instructions under the "Sample Script for Compiling External Libraries and MET" header on this webpage: https://dtcenter.org/community-code/model-evaluation-tools-met/download. I recall there were a few basic Linux tools that were not available in the original environment that I needed to install first. Here is a list of the commands that I ran before following the installation instructions:

> sudo apt-get update
> sudo apt install python-dev
> sudo apt-get install flex bison
> sudo apt-get install libpixman-1-dev
> sudo apt-get install libjpeg-dev
> sudo apt-get install pkg-config
> sudo apt-get install libpng-dev
> sudo apt-get install unzip

Once the MET executables are successfully installed, you can follow the instructions in the METplus User's Guide to obtain the METplus wrapper source code: https://metplus.readthedocs.io/en/latest/Users_Guide/installation.html

You will also have to configure the METplus wrappers to point to the location of the MET executables. There are instructions on the METplus User's Guide here: https://metplus.readthedocs.io/en/latest/Users_Guide/systemconfiguration.html

Please let me know if you run into issues with this process.

I also frequently use Docker to obtain and run METplus, which is a much quicker approach to getting started. The METplus Docker image contains everything you need to run. We also provide Docker data volumes that can be used to run the many example use cases available with the METplus wrappers source code.

I personally do not have any experience using devcontainers in VS Code and I don't think anyone else on the team has either. There is an Amazon AMI that is close to being completed that will be used to run the METplus online tutorial in AWS. If you are interested I can follow up with more information using that when it is available.

**Reply by marstonsward** [reply]
Hi George,

This is fantastic news. Thank you for you detailed reply. While I'm interested in the docker installation. I know a little about docker but I'm eager to learn more. How can I get started with the docker image?
Also, please let me know when the AMI becomes available. We are also interested in this as well.

**Comment by georgemccabe** [comment, marked as answer]
If you have a DockerHub account you can pull the METplus 4.0.0 Docker image with this command:

`docker pull dtcenter/metplus:4.0.0`

You can also pull the latest tag to get the latest version available, which is currently 4.0.0:

`docker pull dtcenter/metplus:latest`

A full list of the available tags can be found here:

https://hub.docker.com/repository/docker/dtcenter/metplus/tags

The sample input data is available as Docker data volumes in the dtcenter/metplus-data DockerHub repository. Each data volume is named after the major/minor version of METplus, i.e. 4.0, and the corresponding set of use cases. The use cases found in parm/use_cases/met_tool_wrapper in the METplus repository use the met_tool_wrapper data and each directory under parm/use_cases/model_applications has a data volume with the same name.

To create a Docker data volume named met_tool_wrapper with the met_tool_wrapper data used for 4.0, run this command:

`docker create --name met_tool_wrapper dtcenter/metplus-data:4.0-met_tool_wrapper`

To start running a Docker container in interactive mode with the data available, run this command:

`docker run --rm -it --volumes-from met_tool_wrapper dtcenter/metplus:4.0.0 bash`

The command line prompt should change if this was successful. To run a use case, simply call the run_metplus.py script (which is already in the path) passing in one of the configuration (.conf) files in the repository:

`run_metplus.py /metplus/METplus/parm/use_cases/met_tool_wrapper/GridStat/GridStat.conf`

All of the use cases can be run using the corresponding input data unless the use case has additional Python package dependencies. This information should be listed in the documentation for the use cases, which can be found in the METplus User's Guide here: https://metplus.readthedocs.io/en/latest/Users_Guide/usecases.html

Please let me know if you have any questions or run into any issues.

**Reply by marstonsward** [reply]
Thank you very much.

I was thinking that the thought was to jump into the image and use it there. I can mount an external volume with my data when I'm ready to test it.
I'll give you some feedback when I have it installed and have run a test case.

**Reply by marstonsward** [reply]
Hi George,

The docker image worked flawlessly. You guys did a great job with this software. The test case you suggested ran without problem.
Now it's up to me to explore this tool and see how to setup my own use case.

Thanks a million,
/M

**Reply by georgemccabe** [reply]
Great to hear! I am going to mark this discussion as answered and lock it, but feel free to open a new discussion topic if you have any other questions or issues.
